MongoDbGenericRepository/DefaultBaseMongodbRepository.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Collections.Generic;
+using MongoDB.Driver;
+using System.Threading.Tasks;
+using System.Linq.Expressions;
+using Mongodb.Driver.Extensions.Abstractions;
+using MongoDbGenericRepository;
+using MongoDbGenericRepository.Models;
+
+namespace Mongodb.Driver.Extensions
+{
+    public abstract class DefaultBaseMongodbRepository<TDocument> : BaseMongoRepository, IDefaultBaseMongodbRepository<TDocument>
+        where TDocument : IDocument, new()
+    {
+        protected abstract string MongodbConnectionString { get; }
+        protected virtual string DefaultCollectionName => GetDefaultCollectionName<TDocument>();
+
+        #region ctor
+
+        protected DefaultBaseMongodbRepository()
+        {
+            InitRepository();
+        }
+
+        /// <summary>
+        ///     The constructor taking a connection string and a database name.
+        /// </summary>
+        /// <param name="connectionString">The connection string of the MongoDb server.</param>
+        /// <param name="databaseName">The name of the database against which you want to perform operations.</param>
+        protected DefaultBaseMongodbRepository(string connectionString, string databaseName = null) : base(
+            connectionString, databaseName)
+        {
+        }
+
+        /// <summary>
+        ///     The contructor taking a <see cref="IMongoDbContext" />.
+        /// </summary>
+        /// <param name="mongodbContext">A mongodb context implementing <see cref="IMongoDbContext" /></param>
+        protected DefaultBaseMongodbRepository(IMongoDbContext mongodbContext) : base(mongodbContext)
+        {
+        }
+
+        /// <summary>
+        ///     The contructor taking a <see cref="IMongoDatabase" />.
+        /// </summary>
+        /// <param name="mongodbDatabase">A mongodb context implementing <see cref="IMongoDatabase" /></param>
+        protected DefaultBaseMongodbRepository(IMongoDatabase mongodbDatabase) : base(mongodbDatabase)
+        {
+        }
+
+
+        protected void InitRepository()
+        {
+            if (string.IsNullOrEmpty(MongodbConnectionString))
```

Comment:
I disagree with this. MongodbConnectionString is readonly and needs to be passed in ctor,

Reply:
Why is it necessary to construct BaseMongodbRepository and carry connectionString and databaseName every time
This is the reason why I use InitRepository to implement parameterless construction